File: iossdkdemo/SDKDemo/APIControllers/DownloadMediaViewController.m
//
//  DownloadMediaViewController.m
//  SDKDemo
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

#import "DownloadMediaViewController.h"
#import <OPGSDK/OPGSDK.h>


@interface DownloadMediaViewController ()

@end

@implementation DownloadMediaViewController

@synthesize mediaId,mediaType;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.mediaType.delegate=self;
    self.mediaId.delegate=self;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - IBAction
-(IBAction)downloadMedia:(id)sender
{
    OPGSDK* sdk = [OPGSDK new];                   // Creating OPGSDK instance
    NSError *error;
    OPGDownloadMedia *obj = [sdk downloadMediaFile:self.mediaId.text mediaType:self.mediaType.text error:&error];
    
    if ([obj.isSuccess intValue]==1 && error==nil)
    {
        UIAlertController * alert = [UIAlertController
                                     alertControllerWithTitle:@"OPGSDKv2.0"
                                     message:@"Media Downloaded Successfully!"
                                     preferredStyle:UIAlertControllerStyleAlert];
        
        UIAlertAction* okButton = [UIAlertAction
                                   actionWithTitle:@"OK"
                                   style:UIAlertActionStyleDefault
                                   handler:^(UIAlertAction * action) {
                                       //Handle your yes please button action here
                                   }];
        
        [alert addAction:okButton];
        [self presentViewController:alert animated:YES completion:nil];
    }
    else
    {
        UIAlertController * alert = [UIAlertController
                                     alertControllerWithTitle:@"OPGSDKv2.0"
                                     message:obj.statusMessage
                                     preferredStyle:UIAlertControllerStyleAlert];
        
        UIAlertAction* okButton = [UIAlertAction
                                   actionWithTitle:@"OK"
                                   style:UIAlertActionStyleDefault
                                   handler:^(UIAlertAction * action) {
                                       //Handle your yes please button action here
                                   }];
        
        [alert addAction:okButton];
        [self presentViewController:alert animated:YES completion:nil];
    }
}

#pragma mark - TextField delegates
-(BOOL) textFieldShouldReturn:(UITextField *)textField{
    
    [textField resignFirstResponder];
    return YES;
}
@end

File: iossdkdemo/SDKDemo-Swift/APIControllers-Swift/DownloadMediaViewController.swift
//
//  DownloadMediaViewController.swift
//  SDKDemoApp
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

import UIKit

class DownloadMediaViewController: UIViewController, UITextFieldDelegate
{

    @IBOutlet weak var txtMediaID: UITextField!
    @IBOutlet weak var txtMediaType: UITextField!
    var obj : OPGDownloadMedia?
    override func viewDidLoad() {
        super.viewDidLoad()
        txtMediaType.delegate=self
        txtMediaID.delegate=self
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {   //delegate method
        textField.resignFirstResponder()
        return true
    }
    
    @IBAction func downloadMedia(_ sender: AnyObject)
    {
        let sdk = OPGSDK()        // Creating OPGSDK instance
        
        do {
            obj = try sdk.downloadMediaFile(txtMediaID.text, mediaType: txtMediaType.text)
        }
        catch{
            print("Download Media Failed")         /* @"Error Occured. Contact Support!" */
            
        }
        
        if (obj?.isSuccess.int32Value==1)
        {
            let alertController = UIAlertController(title: "OPGSDKv2.0", message: "Media Download Successful!", preferredStyle: .alert)
            
            let defaultAction = UIAlertAction(title: "OK", style: .default, handler: nil)
            alertController.addAction(defaultAction)
            
            present(alertController, animated: true, completion: nil)
        }
        else
        {
            let alertController = UIAlertController(title: "OPGSDKv2.0", message: "Media Download Failed!", preferredStyle: .alert)
            
            let defaultAction = UIAlertAction(title: "OK", style: .default, handler: nil)
            alertController.addAction(defaultAction)
            
            present(alertController, animated: true, completion: nil)
        }
    }


}
